source/common/common/matchers.cc

```diff
@@ -0,0 +1,102 @@
+#include "common/common/matchers.h"
+
+#include "envoy/api/v2/core/base.pb.h"
+
+#include "common/config/metadata.h"
+#include "common/protobuf/protobuf.h"
+
+#include "absl/strings/match.h"
+
+namespace Envoy {
+namespace Matchers {
+
+bool DoubleMatcher::match(double value) const {
+  switch (matcher_.match_pattern_case()) {
+  case envoy::type::matcher::DoubleMatcher::kRange:
+    return matcher_.range().start() <= value && value < matcher_.range().end();
```

Comment:
The docs don't specify whether the ranges are inclusive or exclusive, and it seems a bit odd to to inclusive at start but not at end? Any reason for this? Either way please clearly document.

Reply:
I did this to follow the current [Int64Range](https://github.com/envoyproxy/envoy/blob/42a9cae38dd3bf5264fb83f64e310b0d76ea4257/api/envoy/type/range.proto#L14), The document mentioned this in the definition of `DoubleRange`, also added it to `DoubleMatcher`.